docs/modules/ROOT/pages/join-two-streams.adoc

```diff
@@ -0,0 +1,391 @@
+= Stream-To-Stream Joins with SQL
+:page-layout: tutorial
+:page-product: platform
+:page-categories: SQL, Joins, Stream Processing 
+:page-lang:
+:page-enterprise:
+:page-est-time: 10 mins
+:page-beta: true
+:description: Learn how to join two streams and process the results, using SQL.
+
+{description}
+
+== Context
+
+A stream is an ongoing delivery of events. These events are often of the same type. For example, a stream may contain click events on a website.
+
+If you have two or more streams of related data, you can join them together on a related field, process them, and store the result.
+
+== Example Use Case
+
+You have two streams. One stream contains events for new orders and the other contains events for shipped orders. In this tutorial, you'll use SQL to join these two streams and process them to find out which orders have been successfully shipped, how long it took for them to ship, and the warehouse from which they shipped.
+
+The sample streaming data is adapted from this link:https://developer.confluent.io/tutorials/join-a-stream-to-a-stream/ksql.html[tutorial for Confluent Platform].
+
+== Before you Begin
+
+Before starting this tutorial, make sure that you have the following prerequisites:
+
+* link:https://docs.docker.com/compose/install/[Docker Compose]
+* link:https://git-scm.com/book/en/v2/Getting-Started-Installing-Git[Git]
+
+== Step 1. Clone the Project
+
+To set up the project, you need to download the code from GitHub.
+
+. Clone the GitHub repository.
++
+[tabs] 
+====
+HTTPS:: 
++ 
+--
+```bash
+git clone https://github.com/hazelcast-guides/stream-to-stream-joins.git
+cd stream-to-stream-joins
+```
+--
+SSH:: 
++ 
+--
+```bash
+git clone git@github.com:hazelcast-guides/stream-to-stream-joins.git
+cd stream-to-stream-joins
+```
+--
+====
+
+== Step 2. Start the Docker Containers
+
+In this step, you'll use Docker Compose to start all the Docker containers, including a Kafka broker, Hazelcast Platform, and Management Center.
```

Comment:
This may be my lack of understanding but I can't find a dockerfile for docker to run.

Reply:
you just need the docker-compose.yml file. Are you getting errors?

Reply:
I have a copy of a docker-compose.yml from a getting started tutorial but what should be in it and where does it need to be to to run the code?

Reply:
it's available in this branch. You can checkout the branch and run the commands in the tutorial.

Reply:
ok, but when I clone the project from github shouldn't the `docker-compose.yml` be included? I wasn't in the project that I downloaded.

Reply:
Tested with `docker-compose.yml`. 

![image](https://user-images.githubusercontent.com/106963942/187415502-fbb3193b-90d6-4d39-a456-cb103a9b9695.png)

Reply:
it seems the MC images aren't published for 5.2 snapshot or beta. Could you try pulling the latest changes from GitHub and trying again.

Reply:
Sure. Will have another go :)

Reply:
OK, docker seemed to connect correctly this time but I got concerned when changes/files just kept being downloaded:

```
$ docker compose up -d
management-center Pulling
ksqldb-server Pulling
broker Pulling
hazelcast Pulling
schema-registry Pulling
zookeeper Pulling
ksqldb-cli Pulling
131f1a26eef0 Pulling fs layer
16b78ed2e822 Pulling fs layer
e7233e20a08e Pulling fs layer
9b9cbc4e490c Pulling fs layer
f7352f1ba78a Pulling fs layer
e06c925589f1 Pulling fs layer
3a498372ace6 Pulling fs layer
2ea0fca7e4a4 Pulling fs layer
c33c6c0781fc Pulling fs layer
adb7a858bb08 Pulling fs layer
510a24055739 Pulling fs layer
63373038ce40 Pulling fs layer
d8c6373bcce1 Pulling fs layer
e5a5496a7299 Pulling fs layer
01310b7e802e Pulling fs layer
0a7a877b988e Pulling fs layer
904ccf2e80ce Pulling fs layer
9b9cbc4e490c Waiting
f7352f1ba78a Waiting
2ea0fca7e4a4 Waiting
e06c925589f1 Waiting
c33c6c0781fc Waiting
3a498372ace6 Waiting
adb7a858bb08 Waiting
510a24055739 Waiting
63373038ce40 Waiting
d8c6373bcce1 Waiting
0a7a877b988e Waiting
904ccf2e80ce Waiting
e5a5496a7299 Waiting
213ec9aee27d Already exists
5d94a15ccd92 Pulling fs layer
61c471b08ffa Pulling fs layer
e99bb45da038 Pulling fs layer
4f4fb700ef54 Pulling fs layer
bda4a7b3a532 Pulling fs layer
4f4fb700ef54 Waiting
bda4a7b3a532 Waiting
5d94a15ccd92 Waiting
e99bb45da038 Waiting
61c471b08ffa Waiting
131f1a26eef0 Pulling fs layer
16b78ed2e822 Pulling fs layer
e7233e20a08e Pulling fs layer
9b9cbc4e490c Pulling fs layer
f7352f1ba78a Pulling fs layer
e06c925589f1 Pulling fs layer
3a498372ace6 Pulling fs layer
2ea0fca7e4a4 Pulling fs layer
c33c6c0781fc Pulling fs layer
adb7a858bb08 Pulling fs layer
510a24055739 Pulling fs layer
63373038ce40 Pulling fs layer
d8c6373bcce1 Pulling fs layer
e5a5496a7299 Pulling fs layer
f7352f1ba78a Waiting
01310b7e802e Pulling fs layer
0a7a877b988e Pulling fs layer
904ccf2e80ce Pulling fs layer
904ccf2e80ce Waiting
e06c925589f1 Waiting
3a498372ace6 Waiting
d8c6373bcce1 Waiting
e5a5496a7299 Waiting
01310b7e802e Waiting
0a7a877b988e Waiting
63373038ce40 Waiting
9b9cbc4e490c Waiting
2ea0fca7e4a4 Waiting
c33c6c0781fc Waiting
510a24055739 Waiting
adb7a858bb08 Waiting
16b78ed2e822 Downloading [==================>                                ]     657B/1.732kB
16b78ed2e822 Downloading [==================>                                ]     657B/1.732kB
16b78ed2e822 Downloading [==================================================>]  1.732kB/1.732kB
16b78ed2e822 Verifying Checksum
16b78ed2e822 Download complete
16b78ed2e822 Download complete
a9e23b64ace0 Pulling fs layer
38b71301a1d9 Pulling fs layer
a6167da2033d Pulling fs layer
316f193487da Pulling fs layer
ab53ac7b177d Pulling fs layer
ac64979388a9 Pulling fs layer
4fa36883e4d6 Pulling fs layer
f1f5c1327dc0 Pulling fs layer
82ac3bff2516 Pulling fs layer
cd92d894020d Pulling fs layer
4369090b2cee Pulling fs layer
a9e23b64ace0 Waiting
38b71301a1d9 Waiting
a6167da2033d Waiting
316f193487da Waiting
ab53ac7b177d Waiting
ac64979388a9 Waiting
82ac3bff2516 Waiting
4fa36883e4d6 Waiting
cd92d894020d Waiting
4369090b2cee Waiting
f1f5c1327dc0 Waiting
131f1a26eef0 Downloading [>                                                  ]  407.9kB/39.62MB
131f1a26eef0 Downloading [>                                                  ]  407.9kB/39.62MB
131f1a26eef0 Downloading [=>                                                 ]  817.5kB/39.62MB
131f1a26eef0 Downloading [=>                                                 ]  817.5kB/39.62MB
131f1a26eef0 Downloading [=>                                                 ]  1.219MB/39.62MB
131f1a26eef0 Downloading [=>                                                 ]  1.219MB/39.62MB
a9e23b64ace0 Pulling fs layer
38b71301a1d9 Pulling fs layer
a6167da2033d Pulling fs layer
316f193487da Pulling fs layer
ab53ac7b177d Pulling fs layer
ac64979388a9 Pulling fs layer
4fa36883e4d6 Pulling fs layer
f1f5c1327dc0 Pulling fs layer
82ac3bff2516 Pulling fs layer
fba78acf861b Pulling fs layer
b72b3f294684 Pulling fs layer
ab53ac7b177d Waiting
ac64979388a9 Waiting
a9e23b64ace0 Waiting
4fa36883e4d6 Waiting
f1f5c1327dc0 Waiting
82ac3bff2516 Waiting
316f193487da Waiting
38b71301a1d9 Waiting
a6167da2033d Waiting
fba78acf861b Waiting
b72b3f294684 Waiting
2408cc74d12b Pulling fs layer
65c3a4e2d0d1 Pulling fs layer
4f4fb700ef54 Pulling fs layer
ad1543421efc Pulling fs layer
5eabb525c72e Pulling fs layer
6552b0278f58 Pulling fs layer
d5e5942065de Pulling fs layer
813676b10190 Pulling fs layer
0a68573d2ad6 Pulling fs layer
65c3a4e2d0d1 Waiting
4f4fb700ef54 Waiting
ad1543421efc Waiting
5eabb525c72e Waiting
6552b0278f58 Waiting
d5e5942065de Waiting
813676b10190 Waiting
a9e23b64ace0 Pulling fs layer
38b71301a1d9 Pulling fs layer
a6167da2033d Pulling fs layer
316f193487da Pulling fs layer
ab53ac7b177d Pulling fs layer
ac64979388a9 Pulling fs layer
4fa36883e4d6 Pulling fs layer
f1f5c1327dc0 Pulling fs layer
82ac3bff2516 Pulling fs layer
ee297d82fdf3 Pulling fs layer
a9b292ea5ad9 Pulling fs layer
38b71301a1d9 Waiting
a6167da2033d Waiting
a9e23b64ace0 Waiting
316f193487da Waiting
ab53ac7b177d Waiting
ac64979388a9 Waiting
ee297d82fdf3 Waiting
4fa36883e4d6 Waiting
f1f5c1327dc0 Waiting
a9b292ea5ad9 Waiting
82ac3bff2516 Waiting
131f1a26eef0 Downloading [==>                                                ]  1.629MB/39.62MB
131f1a26eef0 Downloading [==>                                                ]  1.629MB/39.62MB
131f1a26eef0 Downloading [==>                                                ]  2.038MB/39.62MB
131f1a26eef0 Downloading [==>                                                ]  2.038MB/39.62MB
9b9cbc4e490c Downloading [>                                                  ]  77.44kB/7.34MB
9b9cbc4e490c Downloading [>                                                  ]  77.44kB/7.34MB
9b9cbc4e490c Downloading [=>                                                 ]  151.4kB/7.34MB
9b9cbc4e490c Downloading [=>                                                 ]  151.4kB/7.34MB
131f1a26eef0 Downloading [===>                                               ]  2.448MB/39.62MB
131f1a26eef0 Downloading [===>                                               ]  2.448MB/39.62MB
9b9cbc4e490c Downloading [==>                                                ]  306.4kB/7.34MB
9b9cbc4e490c Downloading [==>                                                ]  306.4kB/7.34MB
9b9cbc4e490c Downloading [==>                                                ]  388.3kB/7.34MB
9b9cbc4e490c Downloading [==>                                                ]  388.3kB/7.34MB
131f1a26eef0 Downloading [===>                                               ]  2.853MB/39.62MB
131f1a26eef0 Downloading [===>                                               ]  2.853MB/39.62MB
9b9cbc4e490c Downloading [===>                                               ]  470.3kB/7.34MB
9b9cbc4e490c Downloading [===>                                               ]  470.3kB/7.34MB
131f1a26eef0 Downloading [====>                                              ]  3.263MB/39.62MB
131f1a26eef0 Downloading [====>                                              ]  3.263MB/39.62MB
9b9cbc4e490c Downloading [===>                                               ]  551.3kB/7.34MB
9b9cbc4e490c Downloading [===>                                               ]  551.3kB/7.34MB
131f1a26eef0 Downloading [====>                                              ]  3.672MB/39.62MB
131f1a26eef0 Downloading [====>                                              ]  3.672MB/39.62MB
9b9cbc4e490c Downloading [====>                                              ]  715.1kB/7.34MB
9b9cbc4e490c Downloading [====>                                              ]  715.1kB/7.34MB
9b9cbc4e490c Downloading [=====>                                             ]    879kB/7.34MB
9b9cbc4e490c Downloading [=====>                                             ]    879kB/7.34MB
131f1a26eef0 Downloading [=====>                                             ]  4.074MB/39.62MB
131f1a26eef0 Downloading [=====>                                             ]  4.074MB/39.62MB
9b9cbc4e490c Downloading [======>                                            ]  960.9kB/7.34MB
9b9cbc4e490c Downloading [======>                                            ]  960.9kB/7.34MB
9b9cbc4e490c Downloading [=======>                                           ]  1.043MB/7.34MB
9b9cbc4e490c Downloading [=======>                                           ]  1.043MB/7.34MB
131f1a26eef0 Downloading [=====>                                             ]  4.483MB/39.62MB
131f1a26eef0 Downloading [=====>                                             ]  4.483MB/39.62MB
9b9cbc4e490c Downloading [========>                                          ]  1.207MB/7.34MB
9b9cbc4e490c Downloading [========>                                          ]  1.207MB/7.34MB
9b9cbc4e490c Downloading [========>                                          ]  1.289MB/7.34MB
9b9cbc4e490c Downloading [========>                                          ]  1.289MB/7.34MB
e7233e20a08e Downloading [>                                                  ]  535.8kB/231.7MB
e7233e20a08e Downloading [>                                                  ]  535.8kB/231.7MB
131f1a26eef0 Downloading [======>                                            ]  4.893MB/39.62MB
131f1a26eef0 Downloading [======>                                            ]  4.893MB/39.62MB
9b9cbc4e490c Downloading [=========>                                         ]  1.452MB/7.34MB
9b9cbc4e490c Downloading [=========>                                         ]  1.452MB/7.34MB
9b9cbc4e490c Downloading [==========>                                        ]  1.534MB/7.34MB
9b9cbc4e490c Downloading [==========>                                        ]  1.534MB/7.34MB
9b9cbc4e490c Downloading [===========>                                       ]  1.616MB/7.34MB
9b9cbc4e490c Downloading [===========>                                       ]  1.616MB/7.34MB
9b9cbc4e490c Downloading [===========>                                       ]  1.698MB/7.34MB
9b9cbc4e490c Downloading [===========>                                       ]  1.698MB/7.34MB
131f1a26eef0 Downloading [======>                                            ]  5.303MB/39.62MB
131f1a26eef0 Downloading [======>                                            ]  5.303MB/39.62MB
9b9cbc4e490c Downloading [============>                                      ]   1.78MB/7.34MB
9b9cbc4e490c Downloading [============>                                      ]   1.78MB/7.34MB
9b9cbc4e490c Downloading [=============>                                     ]  1.944MB/7.34MB
9b9cbc4e490c Downloading [=============>                                     ]  1.944MB/7.34MB
9b9cbc4e490c Downloading [==============>                                    ]  2.108MB/7.34MB
9b9cbc4e490c Downloading [==============>                                    ]  2.108MB/7.34MB
131f1a26eef0 Downloading [=======>                                           ]    5.7MB/39.62MB
131f1a26eef0 Downloading [=======>                                           ]    5.7MB/39.62MB
9b9cbc4e490c Downloading [===============>                                   ]  2.276MB/7.34MB
9b9cbc4e490c Downloading [===============>                                   ]  2.276MB/7.34MB
e7233e20a08e Downloading [>                                                  ]  1.063MB/231.7MB
e7233e20a08e Downloading [>                                                  ]  1.063MB/231.7MB
9b9cbc4e490c Downloading [================>                                  ]  2.444MB/7.34MB
9b9cbc4e490c Downloading [================>                                  ]  2.444MB/7.34MB
9b9cbc4e490c Downloading [=================>                                 ]  2.607MB/7.34MB
9b9cbc4e490c Downloading [=================>                                 ]  2.607MB/7.34MB
131f1a26eef0 Downloading [=======>                                           ]  6.097MB/39.62MB
131f1a26eef0 Downloading [=======>                                           ]  6.097MB/39.62MB
9b9cbc4e490c Downloading [==================>                                ]  2.771MB/7.34MB
9b9cbc4e490c Downloading [==================>                                ]  2.771MB/7.34MB
9b9cbc4e490c Downloading [===================>                               ]  2.935MB/7.34MB
9b9cbc4e490c Downloading [===================>                               ]  2.935MB/7.34MB
9b9cbc4e490c Downloading [=====================>                             ]  3.099MB/7.34MB
9b9cbc4e490c Downloading [=====================>                             ]  3.099MB/7.34MB
9b9cbc4e490c Downloading [======================>                            ]  3.263MB/7.34MB
9b9cbc4e490c Downloading [======================>                            ]  3.263MB/7.34MB
131f1a26eef0 Downloading [========>                                          ]  6.507MB/39.62MB
131f1a26eef0 Downloading [========>                                          ]  6.507MB/39.62MB
9b9cbc4e490c Downloading [=======================>                           ]  3.423MB/7.34MB
9b9cbc4e490c Downloading [=======================>                           ]  3.423MB/7.34MB
9b9cbc4e490c Downloading [========================>                          ]  3.586MB/7.34MB
9b9cbc4e490c Downloading [========================>                          ]  3.586MB/7.34MB
9b9cbc4e490c Downloading [=========================>                         ]   3.75MB/7.34MB
9b9cbc4e490c Downloading [=========================>                         ]   3.75MB/7.34MB
e7233e20a08e Downloading [>                                                  ]    1.6MB/231.7MB
e7233e20a08e Downloading [>                                                  ]    1.6MB/231.7MB
9b9cbc4e490c Downloading [==========================>                        ]  3.906MB/7.34MB
9b9cbc4e490c Downloading [==========================>                        ]  3.906MB/7.34MB
9b9cbc4e490c Downloading [===========================>                       ]   4.07MB/7.34MB
9b9cbc4e490c Downloading [===========================>                       ]   4.07MB/7.34MB
131f1a26eef0 Downloading [========>                                          ]  6.908MB/39.62MB
131f1a26eef0 Downloading [========>                                          ]  6.908MB/39.62MB
9b9cbc4e490c Downloading [============================>                      ]  4.234MB/7.34MB
9b9cbc4e490c Downloading [============================>                      ]  4.234MB/7.34MB
9b9cbc4e490c Downloading [=============================>                     ]  4.397MB/7.34MB
9b9cbc4e490c Downloading [=============================>                     ]  4.397MB/7.34MB
9b9cbc4e490c Downloading [===============================>                   ]  4.561MB/7.34MB
9b9cbc4e490c Downloading [===============================>                   ]  4.561MB/7.34MB
```

Reply:
this is normal. On Mac you don't get such noisy output

Reply:
So is this the streaming data?

Reply:
No, it's just downloading the Docker images. The first time you run it, it takes a bit longer.

Reply:
We add the streaming data manually in the tutorial.

Reply:
Hi Jake, I gave the process over five minutes this time. But it just looks like a continuous download. Should it take this long?

Reply:
Sorry Jake, I gave the process over 7 minutes this time but the container downloads still didn't complete.

Reply:
@frant-hartm we updated the sample to use the script instead of KSQL and Angela is still having trouble with long builds. It doesn't happen on my machine (Mac), but could it be a Windows issue?

Reply:
No idea.

When I discussed this with Angela she said the bars are still moving. Might it be that she is on a really slow connection? By removing the ksqldb you reduced the size by quite a lot, but it will still download more than 1 GB of data.

She could try to pull the images separately to see which ones is taking that long (`docker pull ...`).